stdlib/public/Concurrency/TaskPrivate.h

```diff
@@ -379,8 +388,10 @@ inline void AsyncTask::flagAsSuspended() {
 
     if (_private().Status.compare_exchange_weak(oldStatus, newStatus,
                                                 std::memory_order_relaxed,
-                                                std::memory_order_relaxed))
+                                                std::memory_order_relaxed)) {
+      swift_task_exitThreadLocalContext((char *)&_private().ExclusivityAccessSet[0]);
```

Comment:
@rjmccall I am pretty sure this is the wrong way to do this. I just put it here for testing purposes. Can you tell me the right place to do this?